dagshub_annotation_converter/ir/image/annotations/base.py

```diff
@@ -0,0 +1,61 @@
+from abc import abstractmethod
+from typing import Optional
+
+from pydantic import BaseModel
+from typing_extensions import Self
+
+from dagshub_annotation_converter.ir.image import NormalizationState
+
+
+class IRAnnotationBase(BaseModel):
+    """
+    Common class for all intermediary annotations
+    """
+
+    filename: Optional[str] = None
+
+    category: str
```

Comment:
> Do you benefit from having this in the base class?
Yes, because I need it if I want to e.g. gather all the classes in the annotations. 

I can turn this into 
```python
@abstractmethod
def get_categories(self) -> list[str]:
    ...
```
and it will work, but I want us to be aware of WHAT this will be used for.

If you want to store confidences like that, how do you expect to convert it to any other format?

> Did you provide a way to have optional custom freestyle information in the annotations?

What for? If that optional information can't be used for conversion, then why have it?
I can add a `meta: dict[str, Any]` into any annotation, but then it's going to be a stringly-typed mess IF we want to utilize it at any point.

I understand your point that the classes here are non-extensible though.

> How hard would that be (including for the backend)?
I gave 0 thought for how this works on the backend, since this isn't interacting with anything for now. What do you mean by backend? I don't expect to be sending this IR to the backend/DE for storage.

Reply:
I mean, in the end this gets converted to LS format and sent to the backend. We shouldn't put anything in there which will prevent LS from reading the task for the common image labeling formats - I'm guessing it will just ignore any additional fields